### PR TITLE
Add new API version 2019-07-01 for SIG swagger spec

### PR DESCRIPTION
**Pull request by jerryc3**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
@AutorestCI regenerate azure-sdk-for-go

**Comment by ravbhatnagar** [comment]
Two properties changed to writable from readonly. Signing off from ARM side.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/examples/CreateOrUpdateASimpleGallery.json
Comment (vanbasten2323) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/examples/CreateOrUpdateASimpleGallery.json]: 2019-07-01? Same for other examples

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/examples/GetAGalleryImageVersion.json
Comment (vanbasten2323) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/examples/GetAGalleryImageVersion.json]: can you give a GET example when source is from snapshots?

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/gallery.json
Comment (vanbasten2323) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/gallery.json]: nit: can you move "GalleryArtifactVersionSource" section to right after "GalleryImageVersionStorageProfile" section, to be consistent?

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/examples/CreateOrUpdateASimpleGallery.json
Comment (jerryc3) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/examples/CreateOrUpdateASimpleGallery.json]: good catch

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/examples/GetAGalleryImageVersion.json
Comment (jerryc3) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/examples/GetAGalleryImageVersion.json]: added

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/gallery.json
Comment (jerryc3) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/gallery.json]: done

**Comment by vanbasten2323** [comment]
LGTM

**Comment by jhendrixMSFT** [comment]
@jerryc3 this looks good from SDK perspective.  Does anybody else need to review/sign off?

**Comment by jerryc3** [comment]
> @jerryc3 this looks good from SDK perspective. Does anybody else need to review/sign off?

Hi Joel, Thanks for the review! I think the code owners Pengfei/Hyonho still need to take a look. I also got informed yesterday that my team is trying to add another small field. I think it would be better to add it to this PR so SDK team only has to regenerate once - I will update this PR as soon as it is ready.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/gallery.json
Comment (vanbasten23) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/gallery.json]: nit: do you think if it'd better to make the description consistent to crp's one?

**Comment by vanbasten23** [comment]
LGTM

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/gallery.json
Comment (jerryc3) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/gallery.json]: Thanks for bringing this up. I checked and the crp one is:
    "HyperVGenerationType": {
      "type": "string",
      "description": "Specifies the HyperVGeneration Type for the Virtual Machine created from the image",
so I think the DiskRP one is better as the name is the same and the description is more general in that it does not imply this resource is a VM.

**Comment by jhendrixMSFT** [comment]
@jerryc3 is this ready now?

**Comment by jerryc3** [comment]
> @jerryc3 is this ready now?

This is ready! I have asked Hyonho to look at it and then we can merge as soon as he signs off.

**Comment by hyonholee** [comment]
LGTM.  Thanks.

**Comment by jerryc3** [comment]
@jhendrixMSFT Got all required approvals. Thanks!

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**
